gui/projectfiledialog.cpp

```diff
@@ -461,15 +461,11 @@ QStringList ProjectFileDialog::getDefines() const
 
 QStringList ProjectFileDialog::getUndefines() const
 {
-    QString undefine = mUI.mEditUndefines->text();
+    QString undefine = mUI.mEditUndefines->text().trimmed();
     QStringList undefines;
-    if (!undefine.isEmpty()) {
-        undefine = undefine.trimmed();
-        if (undefine.indexOf(';') != -1)
-            undefines = undefine.split(";");
-        else
-            undefines.append(undefine);
-    }
+    if (!undefine.isEmpty())
```

Comment:
do you really need to check if `undefine` is not empty here? I assume `QString::split` returns an empty list if the string is empty. How about this code:

    const QString undefine = mUI.mEditUndefines->text().trimmed();
    return undefine.split(";");

If you see such code somewhere else then please tell me... I would not be surprised if we can clean up some of our existing code.

Reply:
My concern is input lik "UNDEF1;  ;". In this case split(";") produces list entry contains whitespace only string. I am not sure that such case is correct for preprocessor. Probably it is needed to removed such entries.

Reply:
that is a good point. Feel free to fix that. I would guess that it can be avoided automatically with input validation.

Reply:
Good result gives
```c++
QStringList ProjectFileDialog::getUndefines() const
{
    QString undefine = mUI.mEditUndefines->text().trimmed();
    QStringList undefines = undefine.split(QRegExp(";\\s*"), QString::SkipEmptyParts);
    return undefines;
}
```
Adding ```undefines.removeDuplicates()```  also can be useful.

Reply:
nice!

Reply:
The input validation could also enforce that the text only contains characters `[a-zA-Z_0-9;]`. I like when the GUI prevents bad input immediately.

Reply:
> The input validation could also enforce that the text only contains characters [a-zA-Z_0-9;]. I like when the GUI prevents bad input immediately.

hmm.. I thought you could configure a QRegExpValidator in the ui file. it seems it must be added in the source code. please add it.